client_sdk/go/sample/sample_connection.yaml

```diff
@@ -0,0 +1,110 @@
+name: test-network-org2
+version: 1.0.0
+client:
+  organization: Org2
+  connection:
+    timeout:
+      peer:
+        endorser: '300'
+organizations:
+  Org2:
+    mspid: Org2MSP
+    peers:
+      - peer0.org2.example.com
+    certificateAuthorities:
+      - ca.org2.example.com
+    cryptoPath: /project/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/test-network/fabric-samples/test-network/organizations/peerOrganizations/org2.example.com/msp
+    users:
+      Admin:
+        cert:
+          path: /project/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/test-network/fabric-samples/test-network/organizations/peerOrganizations/org2.example.com/users/Admin@org2.example.com/msp/signcerts/Admin@org2.example.com-cert.pem
+        key:
+          path: /project/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/test-network/fabric-samples/test-network/organizations/peerOrganizations/org2.example.com/users/Admin@org2.example.com/msp/keystore/priv_sk
+peers:
+  peer0.org2.example.com:
+    url: grpcs://localhost:9051
+    tlsCACerts:
+      pem: |
+        -----BEGIN CERTIFICATE-----
+        MIICWDCCAf6gAwIBAgIRAJNPOYGJsy9KS5pXHfE3YqAwCgYIKoZIzj0EAwIwdjEL
+        MAkGA1UEBhMCVVMxEzARBgNVBAgTCkNhbGlmb3JuaWExFjAUBgNVBAcTDVNhbiBG
+        cmFuY2lzY28xGTAXBgNVBAoTEG9yZzIuZXhhbXBsZS5jb20xHzAdBgNVBAMTFnRs
+        c2NhLm9yZzIuZXhhbXBsZS5jb20wHhcNMjEwNDE0MDg1OTAwWhcNMzEwNDEyMDg1
+        OTAwWjB2MQswCQYDVQQGEwJVUzETMBEGA1UECBMKQ2FsaWZvcm5pYTEWMBQGA1UE
+        BxMNU2FuIEZyYW5jaXNjbzEZMBcGA1UEChMQb3JnMi5leGFtcGxlLmNvbTEfMB0G
+        A1UEAxMWdGxzY2Eub3JnMi5leGFtcGxlLmNvbTBZMBMGByqGSM49AgEGCCqGSM49
+        AwEHA0IABO0f9uT0L5fwY5N6kFHgWF5qjKKt0olBWciE9TQCgKJsrx1vnTNLlaQ5
+        VlX2sWaDUbPMkFdJ2YcyiydqojVFu7+jbTBrMA4GA1UdDwEB/wQEAwIBpjAdBgNV
+        HSUEFjAUBggrBgEFBQcDAgYIKwYBBQUHAwEwDwYDVR0TAQH/BAUwAwEB/zApBgNV
+        HQ4EIgQgWmk0wetKtC4tSNd1B6a3+pAQnPu+yx8zC8Ys3GSrQnAwCgYIKoZIzj0E
+        AwIDSAAwRQIgDybw8Bn6AAc5curctlFjhyOVX1zsrQUiBk3wgfZc5uYCIQCQH1nV
+        +cEVbS6SoV2jlB0K026S+kJWXzZOf8ThVyVMvA==
+        -----END CERTIFICATE-----
+    grpcOptions:
+      ssl-target-name-override: peer0.org2.example.com
+      hostnameOverride: peer0.org2.example.com
+  peer0.org1.example.com:
+    url: grpcs://localhost:7051
+    tlsCACerts:
+      pem: |
```

Comment:
I am a bit confused here.
The main reason for the connection profile that I see here is to specify the certificate of the peer endpoint. As we are using TLS, this will allow the handshake to succeed.
Two questions:
1. (I guess we quickly talked about this) in theory the client should know the currently valid root certificates in the channel and, additionally, the client should be able to extract the name/id (e.g., `peer0.org2.example.com`) from the certificate received by the server (i.e., `peer0`, following the previous case). Isn't there any such mechanism?
2. the `url` field seems to indicate a sort of DNS translation. Shouldn't this occur as the client tries to connect to (say) `peer0.org1.example.com`? Or perhaps `peer0.org1.example.com` is really a string and not a valid url?

Reply:
The main reason to have the connection profile here is to provide an example, so the FPC Client SDK end-users can easily see what "minimum" configuration is required to use our SDK (Lifecycle and Gateway).

1) In theory this sounds correct. In practise you need to specify at least (that is my interpretation) one peer in the connection profile in order to connect to the network. With the help of that peer you (I mean the Fabric SDK) can then leverage discovery and get endpoints and TLS certs to connect to other peers. All the existing sample configuration files I saw for the Fabric SDK where using this form of configuration - I haven't see a configuration where the client just uses the org root cert to validate peer certs. Maybe there is such a mechanism but I haven't find it yet. @andrew-coleman WDYT?

2) In the Fabric SDK configuration, `peer0.org1.example.com:` is a key string that maps to an entry in the `peers:` section. In this example we just use `url: grpcs://localhost:9051` as I took the connection from the test-network, which uses docker-compose to run the network on your local machine.